packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1713,14 +1715,15 @@ export class Bucket extends BucketBase {
 
     this.disallowPublicAccess = props.blockPublicAccess && props.blockPublicAccess.blockPublicPolicy;
     this.accessControl = props.accessControl;
+    this.objectOwnership = props.objectOwnership;
 
     // Enforce AWS Foundational Security Best Practice
     if (props.enforceSSL) {
       this.enforceSSLStatement();
     }
 
     if (props.serverAccessLogsBucket instanceof Bucket) {
-      props.serverAccessLogsBucket.allowLogDelivery();
+      props.serverAccessLogsBucket.allowLogDelivery(this, Stack.of(this).account, props.serverAccessLogsPrefix);
```

Comment:
This is where we setup the log delivery policy/acl, but it only gets done if the user passes in a bucket to `serverAccessLogsBucket`. If a user provides `serverAccessLogsPrefix` but does not provide `serverAccessLogsBucket`, then access logs are sent to the source bucket. Does it need to be something like

```ts
if (props.serverAccessLogsBucket instanceof Bucket) {
  ...
} else if (props.serverAcessLogsPrefix && !props.serverAccessLogsBucket) {
  this.allowLogDelivery(this, Stack.of(this).account, props.serverAccessLogsPrefix);
}
```

It's not being done today so it could be a bug or it could be that you don't need to grant access to allow access logs to be written to the same bucket?